HW0/P5/P5.txt

```diff
@@ -1 +1,21 @@
 # Your answers here.
+
+#1. It would take the cashier 255 seconds, because it takes 255 times to sum up all the 256 numbers if add two numbers at a time, and it takes one second to add two numbers.
+
+#2. They can verify the total sum faster than the original cashier. They can do this in 34 seconds. The fastest way is for each cashier to verify the same number of bags: 256/8 = 32, it takes all of them 31 seconds. Then it takes 3 seconds to add the 8 numbers. In total, it takes 34 seconds.
+
+#3. If the store has an infinite number of employees, it takes 8 seconds to verify the sum. First, all 256 bags are verified using 1 second. Then add two numbers at a time, there are 128 counts using another 1 second. Continue this process, it takes 8 steps to add all numbers, which uses 8 seconds.
+
+#4. Suppose the time to count total is x, for these N bags, we have 2^(x) = N if adding two bags at a time. Thus the time to count is log2(N). 
+
+#5. If we also consider the time to communicate, the answer would be the time in #3 plus communication time, which is 8+7 = 15 seconds, since while adding two numbers, it also takes 1 second to communicate.
+
+#6. If the each cashier only takes 1 bag, it will take 256+5 = 261 seconds, since it takes 1 second to hand bags to each cashier, then plus the counting time. When handing bag to cashiers, the cashiers who already got the bags can start counting, thus it save some time. 
```

Comment:
Thank you for introducing this piazza post! I totally agree with you and the post. I think N does mean the owner can give a maximum of N bags per second. In this question, we have to balance the counting-time and handing-bag time, and while distributing bags, cashiers could start counting and communicating at the same time.